### PR TITLE
change preload_rsc value to set right before resources are sent

### LINKED ISSUE #11780: RESOURCE_URLs / preload_rsc does not work
**Problem Description**:
The RESOURCE_URLs config setting that is meant to enable web delivery of the game's cached RSC file, does not work.

**What did you expect to happen**: 
1) Set RESOURCE_URLS http://nanotrasen.se/paradise.rsc.zip in config.txt
2) Clear byond cache
3) Fire up 'Resource Monitor' windows program, switch to 'overview' then in the 'network' box, sort by 'Receive (B/sec)'.
4) Re-open BYOND, reconnect to game server
5) Observe that there is a 100M download in approximately 10 seconds from an IP that is different to the game server (IE: is cloudfront/website)
6) Successfully connect to the game faster than normal

**What happened instead**:
During step 5, the game downloads the .rsc or .rsc.zip from the actual game server, then initiates a completely unnecessary second download, in the background, re-downloading the same file from the website

**Why is this bad/What are the consequences:**
I want to use RESOURCE_URLS / preload_rsc to have clients download the .rsc or .rsc.zip from a web server instead of the game server. Currently, that doesn't appear to be possible - it downloads twice (game server first) which is slower and negates the point of the exercise.
This is bad because I suspect that serving >100MB file downloads to clients who connect midround is a major cause of lag spikes during normal play for everyone currently connected.

**Steps to reproduce the problem**:
See above

**When did the problem start happening**:
Unknown, possibly broken for years

**Extra Information**:
In the code, if you change:
 preload_rsc = 1 // This is 0 so we can set it to an URL once the player logs in and have them download the resources from a different server.    
To:
preload_rsc = "http://nanotrasen.se/paradise.rsc.zip"

And comment out this block:
if(config.resource_urls)
 preload_rsc = pick(config.resource_urls)
else preload_rsc = 1 // If config.resource_urls is not set, preload like normal.

Then connect to the server, it will download the rsc/rsc.zip from the correct web URL.... but it never actually launches the game.

So, right now, the .rsc/.rsc.zip files are there, regenerated with every server code compile, but the game code itself doesn't seem to support actually using them, which is a shame. If we could start using them, initial download would be faster (up to 20x faster) and it might help resolve some of our lag issues with clients connecting midround.

Related:
Explanation of preload_rsc values (Byond docs): 
preload_rsc_explanation

TG's system: 
https://github.com/tgstation/tgstation/search?q=preload_rsc&unscoped_q=preload_rsc

Explanation of TG's system:
https://github.com/tgstation/tgstation/blob/master/README.md
(scroll to: "WEB/CDN RESOURCE DELIVERY")

### PR DESCRIPTION
**Pull request by Couls**
**What does this PR do:**
fixes #11780, didn't see any increased network usage from my BYOND server while joining after cache is cleared, may need to test to see if this fixes it on the live server as well

**Changelog:**
:cl:
fix: issue with preload_rsc
/:cl:

### CODE REVIEW

**Comment by Kyep** [comment]
Here is how you test this:

Add the following to your config.txt
## Download the RSC from the web
#RESOURCE_URLS http://nanotrasen.se/paradise.rsc.zip

Clear your cache, quit byond.

Connect from a server that is NOT localhost.

Open 'resource monitor' on your local machine.

Sort by Receive (B/sec).

Check whether it does the 100M initial RSC download from the game server IP, the website IP, or twice (once from the game server then again from the web server - this is current behavior).

Assuming it only downloads from the webserver (thanks to this PR, and the config...) then you play the game for a few minutes and verify everything actually works.

Yes, its a lot of testing, but its the only method I found when working on this myself, to see what it was REALLY doing.

**Comment by Couls** [comment]
Got it working

**Comment by AffectedArc07** [comment]
So this is why it never worked when I tested this an age ago. I thought it was cloudflare not caching properly, but turns out is BYOND

Why am I noticing a pattern with BYOND being shit

File: code/modules/client/client procs.dm [hunk 1]
```diff
@@ -755,6 +750,10 @@
 
 //Send resources to the client.
 /client/proc/send_resources()
+	// Change the way they should download resources.
+	if(config.resource_urls)
+		preload_rsc = pick(config.resource_urls)
+	else preload_rsc = 1 // If config.resource_urls is not set, preload like normal.
```
Comment (Fox-McCloud) [inline code/modules/client/client procs.dm]: fix this, please.

Comment (Couls) [reply code/modules/client/client procs.dm]: fixed

**Pull request merged**